examples/multimodel/modular_speechllm/eval_audio_gpt_lora.py

```diff
@@ -0,0 +1,213 @@
+from nemo.collections.nlp.models.language_modeling.megatron_gpt_peft_models import MegatronGPTPEFTModel
+from nemo.collections.nlp.models.language_modeling.megatron_gpt_sft_model import MegatronGPTSFTModel
+from nemo.collections.nlp.models.nlp_model import NLPModel
+from nemo.collections.nlp.parts.nlp_overrides import (
+    GradScaler,
+    MegatronHalfPrecisionPlugin,
+    NLPDDPStrategy,
+    NLPSaveRestoreConnector,
+    PEFTSaveRestoreConnector,
+    PipelineMixedPrecisionPlugin,
+)
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+
+mp.set_start_method("spawn", force=True)
+"""
+This is the script to run inference with a speechllm model.
+
+If you want to evaluate an speechllm file:
+
+python eval_audio_gpt_lora.py \
+	model.restore_from_path=$MEGATRON_CKPT \
+    model.peft.restore_from_path=$ALM_CKPT \
+    model.peft.restore_from_hparams_path=$ALM_YAML \
+    model.data.test_ds.manifest_filepath=$VAL_MANIFESTS \
+    model.data.test_ds.names=$VAL_NAMES \
+    model.data.test_ds.global_batch_size=4 \
+    ++inference.greedy=False \
+    ++inference.temperature=0.8 \
+    model.data.test_ds.micro_batch_size=4 \
+    model.data.test_ds.tokens_to_generate=128
+"""
+
+
+@hydra_runner(config_path="../conf/speechllm", config_name="modularized_speech_gpt_config_eval")
+def main(cfg) -> None:
+    logging.info("\n\n************** Experiment configuration ***********")
+    logging.info(f"\n{OmegaConf.to_yaml(cfg)}")
+    assert cfg.model.restore_from_path is not None
+    megatron_amp_o2 = cfg.model.get("megatron_amp_O2", False)
+    with_distributed_adam = False
+
+    plugins = []
+    strategy = NLPDDPStrategy(
+        no_ddp_communication_hook=True,  # we don't use DDP for async grad allreduce
+        gradient_as_bucket_view=cfg.model.gradient_as_bucket_view,
+        find_unused_parameters=False,
+    )
+    if cfg.trainer.precision in [16, "bf16"]:
+        scaler = None
+        if cfg.trainer.precision == 16:
+            scaler = GradScaler(
+                init_scale=cfg.model.get("native_amp_init_scale", 2 ** 32),
+                growth_interval=cfg.model.get("native_amp_growth_interval", 1000),
+                hysteresis=cfg.model.get("hysteresis", 2),
+                enabled=False
+                if cfg.model.pipeline_model_parallel_size > 1
+                else True,  # turn off the grad scale for pipeline parallel LM model
+            )
+        if megatron_amp_o2 and not with_distributed_adam:
+            plugins.append(MegatronHalfPrecisionPlugin(precision=cfg.trainer.precision, device="cuda", scaler=scaler))
+        else:
+            plugins.append(PipelineMixedPrecisionPlugin(precision=cfg.trainer.precision, device="cuda", scaler=scaler))
+
+    if cfg.get("cluster_type", None) == "BCP":
+        plugins.append(TorchElasticEnvironment())
+
+    trainer = Trainer(plugins=plugins, strategy=strategy, **cfg.trainer)
```

Comment:
You should look into using the MegatronTrainerBuilder as shown in https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/language_modeling/megatron_gpt_pretraining.py#L33C15-L33C37

Reply:
Great comment! Done.